src/components/connect.js

```diff
@@ -88,7 +90,7 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
           this.finalMapStateToProps(state, props) :
           this.finalMapStateToProps(state)
 
-        return checkStateShape(stateProps)
+        return checkStateShape(this.constructor.displayName, stateProps)
```

Comment:
Done. I move `checkStateShape` and also `computeMergedProps` into `wrapWithConnect`.